frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -317,33 +317,54 @@ sendConfig model initData = Workflow $ do
               cannotBeReceiverMsg = "Sender cannot be the receiver of a transfer"
 
-          decoded <- fmap snd $ mkLabeledInput True "Tx Builder" (uiInputWithInlineFeedback
-            (fmap (first (const "Invalid Tx Builder") . Aeson.eitherDecodeStrict . T.encodeUtf8) . value)
-            inputIsDirty
-            T.pack
-            Nothing
-            uiInputElement
-            )
-            (def & inputElementConfig_initialValue .~ (maybe "" (T.decodeUtf8 . LBS.toStrict . Aeson.encode) mInitToAddress))
+              renderTxBuilder = T.decodeUtf8 . LBS.toStrict . Aeson.encode
+              validateTxBuilder = Aeson.eitherDecodeStrict . T.encodeUtf8
+
+              uiTxBuilderInput cfg = do
+                ie <- uiInputElement cfg
+                pure (ie
+                     , ( validateTxBuilder <$> _inputElement_input ie
+                       , validateTxBuilder <$> value ie
+                       )
+                     )
+
+              showTxBuilderPopover (_, (onInput, _)) = pure $ ffor onInput $ \case
+                Left _ ->
+                  PopoverState_Error "Invalid Tx Builder"
+                Right txb ->
+                  if _txBuilder_accountName txb == fromName && _txBuilder_chainId txb == fromChain then
+                    PopoverState_Error cannotBeReceiverMsg
+                  else
+                    PopoverState_Disabled
+
+          decoded <- fmap (snd . snd) $ mkLabeledInput True "Tx Builder"
+            (uiInputWithPopover uiTxBuilderInput (_inputElement_raw . fst) showTxBuilderPopover)
+            (def & inputElementConfig_initialValue .~ (maybe "" renderTxBuilder mInitToAddress))
+
+          let uiGasPriceInput cfg0 = do
+                (ie, amount, onGasPrice) <- mkLabeledInput True "Amount" uiGasPriceInputField cfg0
+                -- We're only interested in the decimal of the gas price
+                  <&> over (_2 . mapped . mapped) (\(GasPrice (ParsedDecimal d)) -> d)
+                pure (ie, (amount, onGasPrice))
 
-          displayImmediateFeedback (updated decoded) cannotBeReceiverMsg
-            $ either (const False) (\ka -> _txBuilder_accountName ka == fromName && _txBuilder_chainId ka == fromChain)
+              showGasPriceInsuffPopover (_, (_, onGasPrice)) = pure $ ffor onGasPrice $ \gp0 ->
+                let fromBal = fromAcc ^? account_status
+                      . _AccountStatus_Exists
+                      . accountDetails_balance
+                      . _AccountBalance
 
-          (_, amount, _) <- mkLabeledInput True "Amount" uiGasPriceInputField
-            (def & inputElementConfig_initialValue .~ (maybe "" tshow mInitAmount))
-            -- We're only interested in the decimal of the gas price
-            <&> over (_2 . mapped . mapped) (\(GasPrice (ParsedDecimal d)) -> d)
+                    (GasPrice (ParsedDecimal gp)) = gp0
+                in
+                  if maybe True (gp >) fromBal then
```

Comment:
I'm not sure we should claim insufficient funds if the balance is unknown.

Reply:
Turns out the current spec means the balance is always known to exist, otherwise we wouldn't be able to be in this dialog. Fixing that in another PR